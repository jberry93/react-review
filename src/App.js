import React, { PureComponent } from 'react'
import { Deck } from './components/Deck/Deck'
import { Form } from './components/Form/Form'
import './index.css'

export class FlashCards extends PureComponent {
  constructor () {
    super()
    this.state = {
      cards: [],
      isNewCard: false
    }
  }

  updateCards = cards => {
    this.setState({ cards })
    this.toggleNewCard()
  }

  toggleNewCard = e => {
    this.setState({ isNewCard: !this.state.isNewCard })
  }

  render () {
    return (
      <div>
        <h1>React Cards</h1>
        <button onClick={this.toggleNewCard}>Create Card</button>
        {this.state.isNewCard &&
          <Form
            cards={this.state.cards}
            updateCards={this.updateCards}
          />
        }
        {this.state.cards.length > 0 ? (
          <Deck cards={this.state.cards}/>
        ) : (
          <h1>Sorry, there are no cards to show</h1>
        )}
      </div>
    )
  }
}
