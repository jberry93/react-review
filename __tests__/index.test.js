import React from 'react'
import { shallow } from 'enzyme'
import App from '../pages/index'

describe('index', () => {
  it('renders w/out crashing', () => {
    shallow(<App/>)
  })
})
