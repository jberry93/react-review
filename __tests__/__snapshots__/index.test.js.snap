// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders w/out crashing: Snapshot 1`] = `
<div
  className="jsx-2292233321"
>
  <div
    className="jsx-553751598"
  >
    <h1
      className="jsx-553751598"
    >
      React Review
    </h1>
    <button
      className="jsx-553751598"
      onClick={[Function]}
    >
      Create Card
    </button>
    <h1
      className="jsx-553751598"
    >
      Sorry there are no cards to show
    </h1>
  </div>
</div>
`;
